### PR TITLE
Fix production image compilation 

### PR DESCRIPTION
**Pull request by sirn**
* Remove Circle CI cache as it’s causing NIF to fail to load.
* Simplify `.circleci/config.yml` by using short form for `run:`
* Simplify `.circleci/config.yml` by splitting long scripts into an actual script file.
* Use `make docker` in CI so we have consistent way to build Docker image.
* Improve conditional CI steps using ENV var detection instead of `CIRCLE_PROJECT_USERNAME`.

### CODE REVIEW

**Comment by sirn** [comment]
Note that removing cache cause the build to take longer, but it should result in a more predictable builds (due to no more randomly failed NIFs loading). Some timing statistics:

**With cache:**

* without E2E: ~6m
* with E2E: ~18m

**Without cache:**

* without E2E: ~8m
* with E2E: ~20m

File: .circleci/ci_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+
+## Sanity check
+##
+
+if [ -z "$CIRCLE_GPG_KEY" ] ||
+       [ -z "$CIRCLE_GPG_OWNERTRUST" ] ||
+       [ -z "$GCP_KEY_FILE" ] ||
+       [ -z "$GPG_ACCOUNT_ID" ] ||
```
Comment (unnawut) [inline .circleci/ci_deploy.sh]: Did you mean `$GCP_ACCOUNT_ID`?

File: .circleci/ci_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+       [ -z "$GCP_REGION" ] ||
+       [ -z "$GCP_ZONE" ] ||
+       [ -z "$GCS_CLUSTER_ID" ]; then
```
Comment (unnawut) [inline .circleci/ci_deploy.sh]: `$GCP_CLUSTER_ID`?

**Comment by sirn** [comment]
I’ll cherry-pick this to 1.2

**Pull request merged**